SparseVector/SparseVector/SparseVector.cs

```diff
@@ -0,0 +1,152 @@
+﻿namespace SparseVector;
+
+/// <summary>
+/// Sparse vector represented by dictionary with row, value key-pair
+/// </summary>
+public class Vector
+{
+    private int numOfRows;
+    public int NumOfRows
+    {
+        get => numOfRows;
+        set => numOfRows = value;
+    }
```

Comment:
```suggestion
    public int NumOfRows { get; set; }
```
Это буквально то же самое, разве что к numOfRows возможен доступ изнутри класса в обход геттера/сеттера (впрочем, думаю, что геттер/сеттер всё равно заинлайнится и это будет буквально обращение к полю, в релизной сборке-то точно). Так что делать неавтоматической свойство, работающее как автоматическое, смысла нет.